Sources/StreamChat/AudioPlayer/AudioPlaying.swift

```diff
@@ -0,0 +1,375 @@
+                    value.state = .playing
+                default:
+                    log.debug("\(type(of: self)): No action for transition \(currentPlaybackState) -> \(newValue)")
+                }
+            }
+        }
+
+        playerObserver.addStoppedPlaybackObserver(
+            queue: nil
+        ) { [weak self] playerItem in
+            guard
+                let self = self,
+                let playerItemURL = (playerItem.asset as? AVURLAsset)?.url,
+                let currentItemURL = (player.currentItem?.asset as? AVURLAsset)?.url,
+                playerItemURL == currentItemURL
+            else {
+                return
+            }
+            var newContext = self.context
+            newContext.state = .stopped
+            newContext.currentTime = 0
+            newContext.rate = .zero
+            newContext.isSeeking = false
+            self.context = newContext
+
+            self.delegate?.audioPlayer(self, didUpdateContext: self.context)
+        }
+    }
+
+    /// Provides thread-safe updates for the player's context and makes sure to forward any updates
+    /// to the the delegate
+    private func updateContext(
+        _ newContextProvider: (inout AudioPlaybackContext) -> Void
+    ) {
+        var newContext = context
+        newContextProvider(&newContext)
+        context = newContext
+        delegate?.audioPlayer(self, didUpdateContext: context)
+    }
+
+    /// It's used by the assetPropertyLoader to handle the completion (successful or failed) of duration's
+    /// asynchronous loading.
+    private func handleDurationLoading(
+        _ result: Result<AVURLAsset, AssetPropertyLoadingCompositeError>
+    ) {
+        guard Thread.isMainThread else {
+            DispatchQueue.main.async { [weak self] in
+                self?.handleDurationLoading(result)
+            }
+            return
+        }
+
+        switch result {
+        /// If the assetPropertyLoaded managed to successfully load the asset's duration information
+        /// we update the context with the new information.
+        case let .success(asset):
+            player.replaceCurrentItem(with: .init(asset: asset))
+            updateContext { value in
+                value.duration = asset.duration.seconds
+                value.currentTime = 0
+                value.rate = .zero
+                value.isSeeking = false
+            }
+
+            player.play()
+
+        /// If the assetPropertyLoader failed to load the asset's duration information we update the
+        /// context with the notLoaded state in order to inform the delegate and we log a debug error message
+        case let .failure(error):
+            updateContext { value in
+                value.duration = 0
+                value.currentTime = 0
+                value.rate = .zero
+                value.state = .notLoaded
+                value.isSeeking = false
+            }
+            log.debug(error.localizedDescription)
```

Comment:
Would be nice that you add a subsystem to the log (this function accepts `subsystems: LogSubsystem`)

Reply:
I had it initially but wasn't sure what's our "policy" for the subsystems. What do you thing about the name `audioPlayback`?